diff --git a/absl/base/internal/sysinfo.cc b/absl/base/internal/sysinfo.cc
--- a/absl/base/internal/sysinfo.cc
+++ b/absl/base/internal/sysinfo.cc
@@ -22,6 +22,10 @@
 
 #include "absl/base/attributes.h"
 
+#ifdef __SWITCH__
+#include <switch.h>
+#endif
+
 #ifdef _WIN32
 #include <windows.h>
 #else
@@ -456,3 +460,15 @@
 pid_t GetTID() { return static_cast<pid_t>(_lwp_self()); }
 
+#elif defined(__SWITCH__)
+
+pid_t GetTID() {
+  uint64_t tid = 0;
+  if (R_FAILED(svcGetThreadId(&tid, CUR_THREAD_HANDLE))) {
+    return -1;
+  }
+  const uint64_t folded_tid = tid ^ (tid >> 32);
+  return static_cast<pid_t>(folded_tid %
+                            static_cast<uint64_t>(std::numeric_limits<pid_t>::max()));
+}
+
 #elif defined(__Fuchsia__)
 
 pid_t GetTID() {
   // Use our thread handle as the TID, which should be unique within
diff --git a/absl/base/internal/thread_identity.cc b/absl/base/internal/thread_identity.cc
--- a/absl/base/internal/thread_identity.cc
+++ b/absl/base/internal/thread_identity.cc
@@ -87,4 +87,4 @@
 #if defined(__wasi__) || defined(__EMSCRIPTEN__) || defined(__MINGW32__) || \
-    defined(__hexagon__)
+    defined(__hexagon__) || defined(__SWITCH__)
-  // Emscripten, WASI and MinGW pthread implementations does not support
-  // signals. See
+  // Emscripten, WASI, MinGW and libnx pthread implementations do not support
+  // signal masking. See
