diff --git a/src/dawn/common/SystemUtils.cpp b/src/dawn/common/SystemUtils.cpp
--- a/src/dawn/common/SystemUtils.cpp
+++ b/src/dawn/common/SystemUtils.cpp
@@ -99,6 +99,12 @@
 bool SetEnvironmentVar(const char* variableName, const char* value) {
+#if DAWN_PLATFORM_IS(SWITCH)
+    static_cast<void>(variableName);
+    static_cast<void>(value);
+    return false;
+#else
     if (value == nullptr) {
         return unsetenv(variableName) == 0;
     }
     return setenv(variableName, value, 1) == 0;
+#endif
 }
