diff --git a/third_party/EGL-Registry/src/api/EGL/eglplatform.h b/third_party/EGL-Registry/src/api/EGL/eglplatform.h
--- a/third_party/EGL-Registry/src/api/EGL/eglplatform.h
+++ b/third_party/EGL-Registry/src/api/EGL/eglplatform.h
@@ -54,1 +54,6 @@
+#elif defined(__SWITCH__)
+typedef void *EGLNativeDisplayType;
+typedef void *EGLNativePixmapType;
+typedef void *EGLNativeWindowType;
+
 #elif defined(_WIN32) || defined(__VC32__) && !defined(__CYGWIN__) && !defined(__SCITECH_SNAP__) /* Win32 and WinCE */
